eclair-core/src/main/scala/fr/acinq/eclair/io/ReconnectionTask.scala

```diff
@@ -53,8 +53,8 @@ class ReconnectionTask(nodeParams: NodeParams, remoteNodeId: PublicKey) extends
   startWith(IDLE, IdleData(Nothing))
 
   when(CONNECTING) {
-    case Event(_: PeerConnection.ConnectionResult.ConnectionFailed, d: ConnectingData) =>
-      log.info(s"connection failed, next reconnection in ${d.nextReconnectionDelay.toSeconds} seconds")
+    case Event(failure: PeerConnection.ConnectionResult.Failure, d: ConnectingData) =>
```

Comment:
Good catch!

But I don't think we want to trigger a reconnection on *all* failures, for example `AlreadyConnected` shouldn't trigger a reconnection otherwise we'd enter a loop...maybe we should define a `attemptReconnection` boolean on `ConnectionResult.Failure` or something like it?

Reply:
Or we could make `AlreadyConnected` a `ConnectionResult.Success`, I remember flip-floping on that one.

Reply:
Good point, it may make more sense that way.

Reply:
In any case, a unit test for the `AlreadyConnected` case will be necessary :)

Reply:
Actually I think we should leave it that way.

This handler is for an automated reconnections, that will be attempted if the peer is in `DISCONNECTED` state. So if we have a race here, it would be with a manual reconnect done in parallel (that's the only thing I can think of). In that case, the reconnection task will schedule a new reconnection attempt _with an increased delay_ (i.e. : no infinite loop), but then the `DISCONNECTED`->`CONNECTED` transition will make the reconnection actor move to `IDLE` and cancelling the reconnection attempt.

Reply:
Can't it simply happen because your peer is connecting to you while you're connecting to him?
In that case if they finished connecting to us, we don't want to break that connection by connecting to them after a delay, do we?

I don't feel good scheduling a new connection when our current status is `AlreadyConnected`, it sounds like something isn't right...

Reply:
> we don't want to break that connection by connecting to them after a delay, do we?

We won't, because we will receive the transition in the `WAITING` state and cancel the scheduled reconnection. We will receive that transition for sure, because the only case where we send a `AlreadyConnected` is if we are indeed in the `CONNECTED` state.

> I don't feel good scheduling a new connection when our current status is AlreadyConnected, it sounds like something isn't right...

I agree, but then again, this whole reconnection logic is complicated and I'm afraid to create an obscure regression by making `AlreadyConnected` be a `Success` or creating special handling for that case.

Reply:
> We won't, because we will receive the transition in the WAITING state and cancel the scheduled reconnection. We will receive that transition for sure, because the only case where we send a AlreadyConnected is if we are indeed in the CONNECTED state.

Can you add a test to ensure that?
TBH it's really hard to ensure by just looking at the code.

Reply:
Done with 098feba5d, but it's not very satisfactory, because the behavior of the peer is mocked. I'm not sure it will give you a clear view of what's happening here. In my defense, the other reconnection tests work that way, and assuming that the peer sends its state transitions is not too far fetched.

What this tests prove, is that we do not enter an infinite loop in the `AlreadyConnected` scenario. If for some reason we miss the `Disconnected`->`Connected` transition from the peer, then we have a bigger problem and all connection failures would lead us to an infinite loop (albeit with an exponential backoff delay).